app/controllers/catalog_controller.rb

```diff
@@ -1143,23 +1143,27 @@ def trees_to_replace(trees)
   end
 
   def set_resource_action(st)
-    d = @edit[:new][:dialog_id].nil? ? nil : Dialog.where(:id => @edit[:new][:dialog_id]).first
+    dialog = Dialog.find_by(:id => @edit[:new][:dialog_id]) if @edit[:new][:dialog_id]
+
     actions = [
-      {:name => 'Provision', :edit_key => :fqname},
-      {:name => 'Reconfigure', :edit_key => :reconfigure_fqname},
-      {:name => 'Retirement', :edit_key => :retire_fqname}
+      {:name => 'Provision',   :edit_fqname_key => :fqname,             :edit_workflow_key => :workflow_id},
+      {:name => 'Reconfigure', :edit_fqname_key => :reconfigure_fqname, :edit_workflow_key => :reconfigure_workflow_id},
+      {:name => 'Retirement',  :edit_fqname_key => :retire_fqname,      :edit_workflow_key => :retire_workflow_id}
     ]
+
     actions.each do |action|
+      fqname      = @edit[:new][action[:edit_fqname_key]]
+      workflow_id = @edit[:new][action[:edit_workflow_key]]
```

Comment:
NOTE not sure that this is how we want to do this, just needed something to show the "autotmate or workflows" logic below.